modules/core/src/main/java/org/locationtech/jts/geom/prep/BasicPreparedGeometry.java

```diff
@@ -31,18 +29,43 @@
  * @author Martin Davis
  *
  */
-class BasicPreparedGeometry 
+class BasicPreparedGeometry extends Geometry
   implements PreparedGeometry
 {
   private final Geometry baseGeom;
   private final List representativePts;  // List<Coordinate>
 
   public BasicPreparedGeometry(Geometry geom) 
   {
+    super(geom.getFactory());
     baseGeom = geom;
     representativePts = ComponentCoordinateExtracter.getCoordinates(geom);
   }
 
+  public Geometry copy() {
+    return baseGeom.copy();
+  }
+
+  public void normalize() {
+    baseGeom.normalize();
+  }
+
+  protected Envelope computeEnvelopeInternal() {
+    return baseGeom.getEnvelopeInternal();
+  }
+
+  protected int compareToSameClass(Object o) {
+    return baseGeom.compareTo(o);
+  }
+
+  protected int compareToSameClass(Object o, CoordinateSequenceComparator comp) {
+    return baseGeom.compareTo(o, comp);
+  }
+
+  protected int getSortIndex() {
+    return 0;
```

Comment:
should this delegate to baseGeom?

Reply:
Same as above. It's protected in Geometry

Reply:
wonder if it will cause any issues?

Reply:
i'd at least add a comment/fixme type message

Reply:
It might also work to have subclasses of BasicPreparedGeometry override it with their own sort index.

Reply:
yeah, i don't know enough about it to say one way or another